### PR TITLE
Style MaseryProgress, Update Goals

### PR DESCRIPTION
**Pull request by ConnorBoulais**
This PR does some styling on the MasteryProgress components from my last PR. It also adds the vertical list of learning targets, which is shown next to a specific target's details (as seen in the design figma for this component).

This PR also makes some updates to goals based on what Christian wanted in our last meeting. Parent dueDates are now visible, and deleting goals is now connected to the backend and is fully functional. There is also a confirmation dialogue that appears when the user attempts to delete a goal.

### CODE REVIEW

File: src/Components/MasteryProgress/MasteryProgress.js [hunk 1]
```diff
@@ -43,91 +47,137 @@ const useStyles = makeStyles({
...
+      </div>
+    );
+  }
 
-function TargetGrid(){
-  return (
-    <div className="gridContainer">
-      <div/>
-      <div className="targetGrid">
+  const targetListStyle = {
```
Comment (tddumont) [inline src/Components/MasteryProgress/MasteryProgress.js:121]: Good job figuring out how to reuse the same cards with a different layout!

File: src/Components/MasteryProgress/TargetDetails.js [hunk 2]
```diff
@@ -62,11 +75,14 @@ function Row(props) {
                 <TableBody>
                   {row.tasks.map((taskMastery, index) => (
                     <TableRow key={index}>
+                      {/* <TableCell align="left">
```
Comment (tddumont) [inline src/Components/MasteryProgress/TargetDetails.js:78]: This could be removed if the styling is finalized.

**Review by tddumont — APPROVED** [review]
Your work on the Mastery Progress looks great!

File: src/App.js [hunk 3]
```diff
@@ -182,7 +182,7 @@ function App() {
         <div>
         <Switch>
           <Route component = {LoginComponent} exact path = '/'/>
-          <Route component = {GoalList} exact path = '/goalsscreen'/>
+          <Route component = {GoalList} exact path = '/goals'/>
```
Comment (Bart-Ilg) [inline src/App.js:185]: clean rename

File: src/Components/Goals/Goal.js [hunk 4]
```diff
@@ -99,6 +114,34 @@ function Goal(props){
     }
   });
 
+  const [deleteGoalMutation] = useMutation(DELETE_GOAL, {
+  
+    update(cache) {
```
Comment (Bart-Ilg) [inline src/Components/Goals/Goal.js:119]: maybe lessen spacing?

File: src/Components/Goals/Goal.js [hunk 5]
```diff
@@ -45,6 +51,15 @@ function Goal(props){
     props.editGoalCallback(strip_subgoal_id(current_goal_state));
   }
 
+  const [dialogueOpen, setDialogueOpen] = React.useState(false);
+  const handleOpenDialogue = () => {
+    setDialogueOpen(true);
+  };
+  const handleCloseDialogue = () => {
```
Comment (Bart-Ilg) [inline src/Components/Goals/Goal.js:58]: space between these?

File: src/Components/Goals/Goal.js [hunk 6]
```diff
@@ -238,8 +281,31 @@ function Goal(props){
...
+        onClose={handleCloseDialogue}
+        aria-labelledby="alert-dialog-title"
+        aria-describedby="alert-dialog-description"
+      >
+        <DialogTitle id="alert-dialog-title">{"Delete Goal"}</DialogTitle>
+        <DialogContent>
+          <DialogContentText id="alert-dialog-description">
+            {"Are you sure you want to delete goal \'" + current_goal_state.title + "\'?"}
+          </DialogContentText>
+        </DialogContent>
```
Comment (Bart-Ilg) [inline src/Components/Goals/Goal.js:299]: good code

File: src/Components/MasteryProgress/MasteryProgress.js [hunk 7]
```diff
@@ -43,91 +47,137 @@ const useStyles = makeStyles({
...
+    }
+  }
+
+  function renderHeader(){
+    if(!focusedTarget){
+      return null;
+    } else {
+      return (
+        <div className="targetDetailsHeaderGrid">
+          <div onClick={closeTargetDetails}>
```
Comment (Bart-Ilg) [inline src/Components/MasteryProgress/MasteryProgress.js:167]: good div organization

File: src/Components/MasteryProgress/TargetDetails.js [hunk 8]
```diff
@@ -98,34 +121,23 @@ function TargetDetails(props){
...
-        <div/>
-      </div>
+    <div style={{height: "max-content", width: "100%"}}>
+      <TableContainer component={Paper}>
+        <Table aria-label="collapsible table">
+          <TableHead>
+            <TableRow>
+              <TableCell />
+              <TableCell align="left"><h2 className={classes.objectiveTitle}>Target Item</h2></TableCell>
+              <TableCell align="center"><h2 className={classes.masteryTitle}>Mastery</h2></TableCell>
```
Comment (Bart-Ilg) [inline src/Components/MasteryProgress/TargetDetails.js:131]: good to condense

**Review by Bart-Ilg — APPROVED** [review]
Looks good overall, seems functional and is largely neat and easy to follow. Some small spacing tweaks could be made but nothing thats a big deal

**Pull request merged**